sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,431 @@
+  } else {
+    return {
+      sqlExpression: value["SqlExpression"],
+      sqlParameters: getSqlParametersOrUndefined(value["Parameters"]),
+      compatibilityLevel: getIntegerOrUndefined(value["CompatibilityLevel"]),
+      requiresPreprocessing: getBooleanOrUndefined(value["RequiresPreprocessing"])
+    };
+  }
+}
+
+/**
+ * Represents settable options on a rule
+ */
+export interface RuleOptions {
+  /**
+   * Defines the expression that the rule evaluates. The expression string is interpreted as a SQL92 expression which must evaluate to True or False. Only one between a correlation and a sql expression can be defined.
+   */
+  filter?: SqlFilter | CorrelationFilter;
+
+  /**
+   * The SQL like expression that can be executed on the message should the associated filter apply.
+   */
+  action?: SqlAction;
+}
+
+/**
+ * @ignore
+ * Internal representation of settable options on a rule
+ */
+export interface InternalRuleOptions extends RuleOptions {
+  /**
+   * Name of the rule.
+   */
+  name?: string;
+}
+
+/**
+ * Represents all attributes of a rule entity
+ */
+export interface Rule extends RuleOptions {
+  /**
+   * Name of the rule
+   */
+  ruleName?: string;
+
+  /**
+   * Name of topic
+   */
+  topicName?: string;
+
+  /**
+   * Name of subscription
+   */
+  subscriptionName?: string;
+
+  /**
+   * Created at timestamp
+   */
+  createdAt?: string;
+}
+
+/**
+ * Represents all possible fields on SqlAction
+ */
+export type SqlAction = SqlFilter;
+
+/**
+ * Represents all possible fields on SqlFilter
+ */
+export interface SqlFilter {
+  /**
+   * SQL expression to use.
+   */
+  sqlExpression?: string;
+
+  /**
+   * SQL parameters to the expression
+   */
+  sqlParameters?: SqlParameter[];
+
+  /**
+   * This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20.
+   */
+  compatibilityLevel?: number;
+
+  /**
+   * Boolean value indicating whether the SQL filter expression requires preprocessing
+   */
+  requiresPreprocessing?: boolean;
+}
+
+/**
+ * @ignore RuleResourceSerializer for serializing / deserializing Rule entities
+ */
+export class RuleResourceSerializer implements AtomXmlSerializer {
+  serialize(rule: InternalRuleOptions): object {
+    const resource: { Name: any; Filter: any; Action: any } = {
+      Filter: {},
+      Action: {},
+      Name: rule.name
+    };
+
+    if (rule.filter == undefined) {
+      // Defaults to creating a true filter if none specified
+      resource.Filter = {
+        SqlExpression: "1=1",
+        CompatibilityLevel: 20
+      };
+      resource.Filter[Constants.XML_METADATA_MARKER] = {
+        "p4:type": "SqlFilter",
+        "xmlns:p4": "http://www.w3.org/2001/XMLSchema-instance"
+      };
+    } else {
+      let isSqlFilter: boolean = false;
+
+      const givenFilter: any = rule.filter as any;
+
+      if (givenFilter.sqlExpression != undefined) {
+        isSqlFilter = true;
+      }
+
+      if (isSqlFilter) {
```

Comment:
Nit: You can skip the need for `isSqlFilter` altogether, by replacing the above `if` check with `if (rule.filter.hasOwnProperty('sqlExpression'))`

Reply:
Updated